packages/tasit-action/src/TransactionSubscription.js

```diff
@@ -11,19 +11,46 @@ export class TransactionSubscription extends Subscription {
   #txConfirmations;
   #timeout;
   #lastConfirmationTime;
-
-  constructor(txPromise, provider) {
+  #contract;
+
+  // Contract passing itself to here, enables
+  // TransactionSubscription calling this.#contract._emitErrorEvent
+  // to emit error to the contract catch-all error listener
+  // As that's is the only function called for now, could be better
+  // pass that function as a callback here?
```

Comment:
I do like this suggestion better than passing in the contract

Reply:
You are using `subscribedEventNames` too though, right? But I have a comment in another thread about why you may not need to use that, in which case the comment above about liking this suggestion better still remains.

Reply:
> You are using subscribedEventNames too though, right? 
Oh, yes.

> But I have a comment in another thread about why you may not need to use that, in which case the comment above about liking this suggestion better still remains.
:+1:

Reply:
I've kept the first approach for now, but I've removed the `subscribedEventNames` call. 
I think that we could close this thread because the open question here is already covered by other threads.

Reply:
Agreed! Resolving